### PR TITLE
metronome musicxml features

### PR DESCRIPTION
**Pull request by rvilarl**
Features missing detected by @jaredjj3 in https://github.com/stringsync/vexml/pull/157

Tests modified to show the new features:

reference
svg_Stave Tempo_Test Bravura svg_reference
current
svg_Stave Tempo_Test Bravura svg_current

### CODE REVIEW

File: tests/stave_tests.ts [hunk 1]
```diff
@@ -690,8 +690,8 @@ function drawTempo(options: TestOptions, contextBuilder: ContextBuilder): void {
   y += 150;
 
   drawTempoStaveBar(120, { duration: 'w', bpm: 80 }, 0);
-  drawTempoStaveBar(100, { duration: 'h', bpm: 90 }, 0);
-  drawTempoStaveBar(100, { duration: 'q', bpm: 96 }, 0);
+  drawTempoStaveBar(100, { duration: 'h', dots2: 1, duration2: 'q' }, 0);
```
Comment (ronyeh) [inline tests/stave_tests.ts:693]: In the tests / examples, we should probably always put dots after duration for consistency. It looks weird here to have duration / dots2.  It should be duration / duration2 / dots2.

File: src/stavetempo.ts [hunk 2]
```diff
@@ -2,17 +2,21 @@
...
 export interface StaveTempoOptions {
-  bpm?: number;
+  name?: string;
+  parenthesis?: boolean;
   duration?: string;
   dots?: number;
-  name?: string;
+  bpm?: number;
+  duration2?: string;
+  dots2?: number;
```
Comment (ronyeh) [inline src/stavetempo.ts:19]: Can we add a comment somewhere in this file explaining what duration2 and dots2 is for?

You can explain it is for cases where the tempo marker is X = Y where both X and Y are notes (not BPM numbers).

File: tests/stave_tests.ts (hunk 1 above)
Comment (rvilarl) [reply tests/stave_tests.ts:693]: done

File: src/stavetempo.ts (hunk 2 above)
Comment (rvilarl) [reply src/stavetempo.ts:19]: done

**Comment by rvilarl** [comment]
@jaredjj3 @ronyeh while documenting I realised that bpm should be a string as the bpm can also be specified, for example, as a range. I have modified the test as well to show a range.

svg_Stave Tempo_Test Bravura svg_current

**Comment by ronyeh** [comment]
Have you pushed the changes? I don't see any of the new work.

**Comment by rvilarl** [comment]
> Have you pushed the changes? I don't see any of the new work.

sorry, now pushed

**Comment by ronyeh** [comment]
Sorry to add more work. 

Doesn't bpm make more sense as a number in almost all cases? The API should maybe accept number | string. String is definitely the less common case. 

Then internally, we could convert all incoming numbers to strings at render time. 

Does that make sense?

It just feels weird to make someone type out '120' with quotes when they just want to set the bpm to 120.

**Comment by rvilarl** [comment]
> Sorry to add more work.
> 
> Doesn't bpm make more sense as a number in almost all cases? The API should maybe accept number | string. String is definitely the less common case.
> 
> Then internally, we could convert all incoming numbers to strings at render time.
> 
> Does that make sense?
> 
> It just feels weird to make someone type out '120' with quotes when they just want to set the bpm to 120.

@ronyeh Done, I think that it is better. Thanks!

**Review by ronyeh — APPROVED** [review]

**Comment by jaredjj3** [comment]
Nicely done! Let me know when this is released and I'll weave it into vexml.

**Comment by rvilarl** [comment]
@jaredjj3 already in version alpha.10. Please note that the library without embedded fonts is vexflow-core.js

**Pull request merged**